src/main/java/se/lth/base/server/mail/CustomEmail.java

```diff
@@ -2,64 +2,510 @@
+        // Matches all instances of Mustache wraps
+        Matcher m = Pattern.compile("\\{\\{[A-z]\\w+}}").matcher(template);
+
+        while (m.find()) {
+            String mustacheValue = m.group().substring(2,m.group().length() - 2);
+            String replacement = replacements.get(mustacheValue);
+            System.out.println(replacement);
+            template = template.replaceAll("\\{\\{" + mustacheValue + "}}", replacement == null ? "" : replacement);
+        }
+
+        return template;
+    }
+
+    private String getButton(String link, String text) {
+        Map<String, String> buttonContent = new HashMap<>();
+        buttonContent.put("buttonLink", link);
+        buttonContent.put("buttonText", text);
+        return parseResource(getResource("button-template.html"), buttonContent);
+    }
+
+    // ---------- STANDARD EMAILS BELOW --------------
+    private String getWelcomeSubject() {
+        return singleUser.getFirstName() + ", welcome to CommuteCompanion";
+    }
+
+    private String getWelcomeBody() {
+        String firstName = singleUser.getFirstName();
+        Map<String, String> content = new HashMap<>();
+
+        // Title
+        content.put(TITLE, getWelcomeSubject());
+
+        // Preview
+        String preview = firstName + ", welcome to CommuteCompanion. Please visit our website to find or post trips.";
+        content.put(PREVIEW, preview);
+
+        // Intro
+        String intro = "<p>Hi " + firstName + ",</p>";
+        intro += "<p>Welcome to CommuteCompanion! Please visit our website to post or find your first trip.</p>";
+        content.put(INTRO, intro);
+
+        // Action button
+        content.put(BUTTON, getButton(WEBSITE_LINK, "Get Started Here"));
+
+        // Add recipient
+        String fullName = singleUser.getFirstName() + " " + singleUser.getLastName();
+        recipients.add(new Recipient(fullName, singleUser.getEmail(), Message.RecipientType.TO));
+
+        return parseResource(getResource(EMAIL_TEMPLATE), content);
+    }
+
+    private String getNewPassengerOnTripSubject() {
+        return "New passenger request for your drive to " + driveWrap.getDrive().getStop();
+    }
+
+    private String getNewPassengerOnTripBody() {
+        Map<String, String> content = new HashMap<>();
+        DriveUser lastPassenger = null, driver = null;
+        Timestamp departureTime = driveWrap.getDrive().getDepartureTime();
+        String formattedDepartureTime = new SimpleDateFormat("EEEEE MMMMM d 'at' HH:mm").format(departureTime);
+
+        // Fetch the driver and last passenger added
+        for (DriveUser u : driveWrap.getUsers()) {
+            if (u.isDriver()) {
+                driver = u;
+            }
+
+            lastPassenger = u;
+        }
+
+        // Get the user data
+        UserDataAccess userDao = new UserDataAccess(Config.instance().getDatabaseDriver());
+        assert driver != null;
```

Comment:
Bad practice to use `assert`. It is used by devs to find bugs during development. This code should be triggered by production code, i.e. user action. If for example driver somehow is null, we need to handle that exception (in this case, throw an exception from method).
There are multiple `assert`, won't comment them all but i think most (all?) can be replaced by
```java
if(var == null) {
   throw new IllegalArgumentException ("var is null");
}
```